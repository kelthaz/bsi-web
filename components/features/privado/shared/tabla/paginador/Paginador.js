import PropTypes from 'prop-types';
import { useState } from 'react';

import styles from './paginador.module.scss';

const Paginador = ({numberOfPages, onChange, currentPage}) => {

  const [selectedPage, setSelectedPage] = useState(currentPage);

  const onPageChange = (numPage) => {
    if (numPage === currentPage || numPage > numberOfPages || numPage < 1) {
      return;
    }
    setSelectedPage(numPage);
    onChange(numPage);
  };

  return (
    <ul className={styles.pagination}>
      <li>
        <span role="button" onClick={() => onPageChange(currentPage - 1)} tabIndex={0}>&lt;</span>
      </li>
      {[...Array(numberOfPages)].map((__, num) =>
        // eslint-disable-next-line react/no-array-index-key
        <li key={num} className={selectedPage === (num + 1) ? styles.active : ''}>
          <span role="button" onClick={() => onPageChange(num + 1)} tabIndex={0}>{num + 1}</span>
        </li>
      )}
      <li>
        <span role="button" onClick={() => onPageChange(currentPage + 1)} tabIndex={0}>&gt;</span>
      </li>
    </ul>
  );
};

Paginador.propTypes = {
  numberOfPages: PropTypes.number.isRequired,
  currentPage: PropTypes.number,
  onChange: PropTypes.func
};

Paginador.defaultProps = {
  currentPage: 1,
  onChange: () => {}
};

export default Paginador;
