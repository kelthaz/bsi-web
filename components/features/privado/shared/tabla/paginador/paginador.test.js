import { shallow } from 'enzyme';
import React from 'react';
import Paginador from './Paginador';

describe('Pruebas en el componente Accordion', () => {

  it('Debe de mostrarse correctamente el componente', () => {
    // arrange
    const onChange = jest.fn();
    const props = { numberOfPages: 5, currentPage: 1, onChange };
    // act
    const wrapper = shallow(<Paginador {...props} />);
    // assert
    expect(wrapper).toMatchSnapshot();
  });

  it('No debe llamar la funci칩n onChange', () => {
    // arrange
    const onChange = jest.fn();
    const props = { numberOfPages: 5, currentPage: 1, onChange };
    // act
    const wrapper = shallow(<Paginador {...props} />);
    wrapper.find('span').at(1).simulate('click');
    // assert
    expect(onChange).not.toHaveBeenCalled();
  });

  it('Debe llamar la funci칩n onChange', () => {
    // arrange
    const onChange = jest.fn();
    const props = { numberOfPages: 5, currentPage: 1, onChange };
    // act
    const wrapper = shallow(<Paginador {...props} />);
    wrapper.find('span').at(2).simulate('click');
    // assert
    expect(onChange).toHaveBeenCalled();
  });

  it('Debe avanzar una p치gina', () => {
    // arrange
    const onChange = jest.fn();
    const props = { numberOfPages: 5, currentPage: 1, onChange };
    // act
    const wrapper = shallow(<Paginador {...props} />);
    wrapper.find('span').last().simulate('click');
    // assert
    expect(onChange).toHaveBeenCalledWith(2);
  });

  it('Debe retroceder una p치gina', () => {
    // arrange
    const onChange = jest.fn();
    const props = { numberOfPages: 5, currentPage: 2, onChange };
    // act
    const wrapper = shallow(<Paginador {...props} />);
    wrapper.find('span').first().simulate('click');
    // assert
    expect(onChange).toHaveBeenCalledWith(1);
  });


  // it('Debe de llamarse el setFieldValue con el valor nuevo y campo a cambiar', () => {
  //   // arrange
  //   const event = { target: { value: 100000 } };
  //   const props = { name: 'monto', value, onChange, onBlur, min: 0, max: 10, step: 1 };
  //   const wrapper = shallow(<Slider {...props} />);
  //   // act
  //   wrapper.find('input').simulate('change', event);
  //   // assert
  //   expect(onChange).toHaveBeenCalledTimes(1);
  //   expect(onChange).toHaveBeenCalledWith(event);
  // });
});
