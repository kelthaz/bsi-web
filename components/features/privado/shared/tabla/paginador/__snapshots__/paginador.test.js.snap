// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Accordion Debe de mostrarse correctamente el componente 1`] = `
<ul
  className="pagination"
>
  <li>
    <span
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      ͔
    </span>
  </li>
  <li
    className="active"
    key="0"
  >
    <span
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      1
    </span>
  </li>
  <li
    className=""
    key="1"
  >
    <span
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      2
    </span>
  </li>
  <li
    className=""
    key="2"
  >
    <span
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      3
    </span>
  </li>
  <li
    className=""
    key="3"
  >
    <span
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      4
    </span>
  </li>
  <li
    className=""
    key="4"
  >
    <span
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      5
    </span>
  </li>
  <li>
    <span
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      ͕
    </span>
  </li>
</ul>
`;
